src/seq2seq.ipynb

```diff
@@ -0,0 +1,686 @@
+     "start_time": "2024-03-23T09:33:11.677603Z"
+    }
+   },
+   "id": "82cf8dea8a0ef552",
+   "execution_count": 38
+  },
+  {
+   "cell_type": "code",
+   "outputs": [],
+   "source": [
+    "class Lang:\n",
+    "    def __init__(self, name):\n",
+    "        self.name = name\n",
+    "        self.word2index = {}\n",
+    "        self.word2count = {}\n",
+    "        self.index2word = {0: \"SOS\", 1: \"EOS\"}\n",
+    "        self.n_words = 2  # Count SOS and EOS\n",
+    "\n",
+    "    def addSentence(self, sentence):\n",
+    "        for word in sentence.split(' '):\n",
+    "            self.addWord(word)\n",
+    "\n",
+    "    def addWord(self, word):\n",
+    "        if word not in self.word2index:\n",
+    "            self.word2index[word] = self.n_words\n",
+    "            self.word2count[word] = 1\n",
+    "            self.index2word[self.n_words] = word\n",
+    "            self.n_words += 1\n",
+    "        else:\n",
+    "            self.word2count[word] += 1"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2024-03-23T09:33:12.198217Z",
+     "start_time": "2024-03-23T09:33:12.193116Z"
+    }
+   },
+   "id": "9d67c1ab9f525cc0",
+   "execution_count": 39
+  },
+  {
+   "cell_type": "code",
+   "outputs": [],
+   "source": [
+    "def readLangs(to_spanish=False):\n",
+    "    print(\"Reading lines...\")\n",
+    "\n",
+    "    # Read the cvs file\n",
+    "    file = pd.read_csv(\"../data/cropped_en_es_translation.csv\")\n",
+    "    en = file[\"en_translation\"]\n",
+    "    es = file[\"es_translation\"]\n",
+    "\n",
+    "    # Split every line into pairs\n",
+    "\n",
+    "    # Reverse pairs, make Lang instances\n",
+    "    if to_spanish:\n",
+    "        pairs = list(zip(es, en))\n",
+    "        input_lang = Lang(\"es\")\n",
+    "        output_lang = Lang(\"en\")\n",
+    "    else:\n",
+    "        pairs = list(zip(en, es))\n",
+    "        input_lang = Lang(\"en\")\n",
+    "        output_lang = Lang(\"es\")\n",
+    "        \n",
+    "    for pair in pairs:\n",
+    "        input_lang.addSentence(pair[0])\n",
+    "        output_lang.addSentence(pair[1])\n",
+    "        \n",
+    "    return input_lang, output_lang, pairs"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2024-03-23T09:33:12.278070Z",
+     "start_time": "2024-03-23T09:33:12.271438Z"
+    }
+   },
+   "id": "96ffffae0e32e967",
+   "execution_count": 40
+  },
+  {
+   "cell_type": "code",
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Reading lines...\n",
+      "('In my opinion the European Union should have the right to be informed of the minimum levels of social benefits in the individual Member States ', 'Teniendo esto en mente el Consejo Europeo sugirió un límite del 1,5 para que el déficit de transposición de las directivas se aplicara a todos los Estados miembros ')\n"
```

Comment:
Should we be concerned that this is not a correct translation?

Reply:
That appears in en_es_translation generated by your preprocessing. Could you check whether the translations roughly match the meaning? I do not know whether we randomly assigned the pair or it is an error within europarl-v7 document

Reply:
There seems to be, indeed, an error in the preprocessing. The early translations are correct, but probably due to the fact that we are not eliminating sentences in a pair-wise manner, this causes an offset. I  will fix this before our next meeting and make a PR to change the preprocessing file. This shouldn't affect your PR.